### ISSUE TITLE
[Compilation] QA Notice: "Build triggers severe warnings which indicate it may exhibit random runtime failures"

### ISSUE THREAD

**Issue by ThisNekoGuy** [cemu-project/Cemu#1356]
### Current Behavior

Gentoo has certain default warnings enabled for compilers that tells the user, after installation of packages, that something could cause a program to randomly crash (somehow), and I got this message when building commit ~~https://github.com/cemu-project/Cemu/commit/8508c625407e80a5a7fcb9cf02c5355d018ff64b~~ a05bdb172d198904a76b4b166b43fb31a94a0dd3:
```
 * QA Notice: Package triggers severe warnings which indicate that it
 *            may exhibit random runtime failures.
 * /var/tmp/portage/games-emulation/cemu-2.0_p20240915/work/Cemu-a05bdb172d198904a76b4b166b43fb31a94a0dd3/src/Cafe/OS/libs/ntag/ntag.cpp:512:53: warning: 'memcpy' call operates on objects of type 'NTAGNoftHeader' while the size is based on a different type 'NTAGNoftHeader *' [-Wsizeof-pointer-memaccess]

 * Please do not file a Gentoo bug and instead report the above QA
 * issues directly to the upstream developers of this software.
 * Homepage: https://cemu.info/ https://github.com/cemu-project/Cemu
```

### Expected Behavior

To not receive a warning about the program having the potential for random runtime failures.

### Steps to Reproduce

- Compile commit ~~https://github.com/cemu-project/Cemu/commit/8508c625407e80a5a7fcb9cf02c5355d018ff64b~~ a05bdb172d198904a76b4b166b43fb31a94a0dd3 using Clang 18.1.8 and LLVM libc++

### System Info (Optional)

OS: Gentoo Linux (64-bit, LLVM18 based)
Toolchain: Clang 18.1.8 with LLVM libc++ & glibc

**Comment by goeiecool9999** [comment]
I reviewed the line that your compiler complains about. 8508c625407e80a5a7fcb9cf02c5355d018ff64b fixes a mistake in the code that luckily didn't cause a bug because the before and after value happen to be the same on 64-bit x86. It definitely should be what it was changed to so I think that compiler warning is a false positive. (assuming 8508c625407e80a5a7fcb9cf02c5355d018ff64b is where the warning was introduced)
